### ISSUE TITLE
SignalException diasallows fail methods on any class

### ISSUE THREAD

**Issue by gowabash** [bbatsov/rubocop#2161]
Signal Exception does not care the type of class that uses the method.  This should only fire when using Exception.fail (or a subclass)

EX.

```ruby
class Test
  def fail
    puts "reporting the failure somewhere"
  end
end

t = Test.new
t.fail
```

**Comment by jonas054** [comment]
But `SignalException` doesn't report any offenses for you example! Can you change it to illustrate your point?

**Comment by gowabash** [comment]
Sorry.  We have an additional rule 

``` yaml
Style/SignalException:
  EnforcedStyle: only_raise
```

That appears to be the cause.

**Comment by jonas054** [comment]
Yes, `t.fail` should not have been reported by `Style/SignalException` since there's an explicit receiver.
